### PR TITLE
ICO-2939/JSON-logging-framework

### PR DESCRIPTION
**Pull request by josippapez**
Added GlookoLogging module for logging messages.
Wrote test for GlookoLogging module.
Minor rubocop fixes.

**Format of the message Is viable to change (include/exclude certain parts of the message).**

### CODE REVIEW

**Review by mariodzoic — APPROVED** [review]

File: spec/lib/base/glooko_logging_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+RSpec.describe 'GlookoLogging' do
+  let(:logger_class) { Class.new { include Base::GlookoLogging } }
+
+  it 'logs expected output with json data' do
+    test = expect do
```
Comment (VlatkoGoljevacki) [inline spec/lib/base/glooko_logging_spec.rb]: can we write this using lazy variables?

let(:test_log) do 
  logger_class.new.log('warn', 
                                      {fog: 'test'}, 
                                      __LINE__, 
                                      __method__,
                                       __FILE__) 
end

expect(test_log).to output()...

File: spec/lib/base/glooko_logging_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      logger_class.new.log('warn', {fog: 'test'}, __LINE__, __method__,
```
Comment (VlatkoGoljevacki) [inline spec/lib/base/glooko_logging_spec.rb]: can we also test that the Logger received "warn" when logging the data?

File: spec/lib/base/glooko_logging_spec.rb (hunk 1 above)
Comment (josippapez) [reply spec/lib/base/glooko_logging_spec.rb]: Changed the code to fit the suggested comment

File: spec/lib/base/glooko_logging_spec.rb (hunk 2 above)
Comment (josippapez) [reply spec/lib/base/glooko_logging_spec.rb]: Please check if the latest change includes the requested test

**Review by VlatkoGoljevacki — APPROVED** [review]

**Review by elliothutchinson — COMMENTED** [review]
I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?

**Comment by josippapez** [comment]
> I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?

Yeah, I was thinking of just setting up basic logger for now and when we have finalized log format, we can add/remove data.

**Review by elliothutchinson — APPROVED** [review]

**Comment by VlatkoGoljevacki** [comment]
> > I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?
> 
> Yeah, I was thinking of just setting up basic logger for now and when we have finalized log format, we can add/remove data.

@josippapez Can you create a ticket for this and put in the prioritized backlog so we don't forget about it this sprint?

**Comment by josippapez** [comment]
@VlatkoGoljevacki yes, I'll add it to the backlog

**Review by kolaricmatej — APPROVED** [review]

**Pull request merged**